Issue when running tests locally

When I just setup a fork of main and ran the tests locally (with jax v0.4.38, eqx 0.11.11, lx 0.0.5, optx 0.0.9) I see the following for some of the integrate tests

```
FAILED test/test_integrate.py::test_basic[stepsize_controller0-treedef0-float32-int32-solver5] - jax._src.dtypes.TypePromotionError: Input dtypes ('float64', 'float32') have no available implicit dtype promotion path when jax_n...
E       jax._src.dtypes.TypePromotionError: Input dtypes ('float64', 'float32') have no available implicit dtype promotion path when jax_numpy_dtype_promotion=strict. Try explicitly casting inputs to the desired output type, or set jax_numpy_dtype_promotion=standard.
```

I don't see them on the recent PRs https://github.com/patrick-kidger/diffrax/pull/555, so I was just curious if you knew what could be causing it before I dig deeper into it.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Mathematics",
 ]
 urls = {repository = "https://github.com/patrick-kidger/diffrax" }
-dependencies = ["jax>=0.4.38", "jaxtyping>=0.2.24", "typing_extensions>=4.5.0", "typeguard==2.13.3", "equinox>=0.11.10", "lineax>=0.0.5", "optimistix>=0.0.9"]
+dependencies = ["jax>=0.4.38", "jaxtyping>=0.2.24", "typing_extensions>=4.5.0", "typeguard==2.13.3", "equinox>=0.11.10", "lineax>=0.0.5", "optimistix>=0.0.10"]
 
 [build-system]
 requires = ["hatchling"]

